applications/IgaApplication/custom_elements/laplacian_IGA_element.cpp

```diff
@@ -0,0 +1,231 @@
+// KRATOS ___ ___  _  ___   __   ___ ___ ___ ___
+//       / __/ _ \| \| \ \ / /__|   \_ _| __| __|
+//      | (_| (_) | .` |\ V /___| |) | || _|| _|
+//       \___\___/|_|\_| \_/    |___/___|_| |_|  APPLICATION
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//  Main authors:    Nicolò Antonelli
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/checks.h"
+#include "includes/define.h"
+#include "includes/kratos_flags.h"
+#include "includes/ublas_interface.h"
+#include "includes/variables.h"
+#include "includes/convection_diffusion_settings.h"
+#include "utilities/geometry_utilities.h"
+
+// Application includes
+#include "custom_elements/laplacian_IGA_element.h"
+
+#include "utilities/math_utils.h"
+
+namespace Kratos
+{
+
+LaplacianIGAElement::LaplacianIGAElement(
+    IndexType NewId,
+    GeometryType::Pointer pGeometry)
+    : Element(
+        NewId,
+        pGeometry)
+{
+}
+
+LaplacianIGAElement::LaplacianIGAElement(
+    IndexType NewId,
+    GeometryType::Pointer pGeometry,
+    PropertiesType::Pointer pProperties)
+    : Element(
+        NewId,
+        pGeometry,
+        pProperties)
+{
+}
+
+Element::Pointer LaplacianIGAElement::Create(
+    IndexType NewId,
+    NodesArrayType const& ThisNodes,
+    PropertiesType::Pointer pProperties) const
+{
+    return Kratos::make_intrusive<LaplacianIGAElement>(NewId, GetGeometry().Create(ThisNodes), pProperties);
+}
+
+Element::Pointer LaplacianIGAElement::Create(
+    IndexType NewId,
+    GeometryType::Pointer pGeom,
+    PropertiesType::Pointer pProperties) const
+{
+    return Kratos::make_intrusive<LaplacianIGAElement>(NewId, pGeom, pProperties);
+}
+
+// Deconstructor
+
+LaplacianIGAElement::~LaplacianIGAElement()
+{
+}
+
+
+// From classical Laplacian
+void LaplacianIGAElement::CalculateLocalSystem(MatrixType& rLeftHandSideMatrix,
+                                            VectorType& rRightHandSideVector,
+                                            const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    const ProcessInfo& r_process_info = rCurrentProcessInfo;
+    ConvectionDiffusionSettings::Pointer p_settings = r_process_info[CONVECTION_DIFFUSION_SETTINGS];
+    auto& r_settings = *p_settings;
+
+    const Variable<double>& r_unknown_var = r_settings.GetUnknownVariable(); // Temperature
+    const Variable<double>& r_diffusivity_var = r_settings.GetDiffusionVariable(); // Conductivity
+    const Variable<double>& r_volume_source_var = r_settings.GetVolumeSourceVariable(); // HeatFlux
+
+    // reading integration points and local gradients
+    const auto& r_geometry = GetGeometry();
+    const GeometryType::IntegrationPointsArrayType& integration_points = r_geometry.IntegrationPoints(this->GetIntegrationMethod());
+    const GeometryType::ShapeFunctionsGradientsType& DN_De = r_geometry.ShapeFunctionsLocalGradients(this->GetIntegrationMethod());
```

Comment:
```suggestion
    const GeometryType::IntegrationPointsArrayType& r_integration_points = r_geometry.IntegrationPoints(this->GetIntegrationMethod());
    const GeometryType::ShapeFunctionsGradientsType& r_DN_De = r_geometry.ShapeFunctionsLocalGradients(this->GetIntegrationMethod());
```
Please follow style guide. References must be precluded by the `r_` keyword.